### PR TITLE
Fix compilation with pattern feature

### PR DESCRIPTION
**Pull request by Luro02**

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -21,6 +21,7 @@ features=(
     "std perf-dfa"
     "std perf-inline"
     "std perf-literal"
+    "std pattern"
```
Comment (BurntSushi) [inline test:24]: Unfortunately, this script is only run on stable Rust. So I think you'll have to add

```
cargo test --test default --no-default-features --features 'std pattern'
```

as an explicit step in CI on nightly targets only.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
Thanks! It looks like the CI setup might need to be tweaked a bit though.

File: test (hunk 1 above)
Comment (Luro02) [reply test]: done

**Comment by BurntSushi** [comment]
Looks like the `unicode-perl` feature is needed to run the searcher tests.

GitHub Actions CI is FUBAR, so it doesn't run until I click a button. Please run the tests locally and make sure it works there before pushing.

**Review by BurntSushi — APPROVED** [review]

**Comment by BurntSushi** [comment]
This PR is on crates.io in `regex 1.5.4`.

**Pull request merged**